README.txt

$Id: README.txt,v 1.5 2010-02-03 17:45:53 tkralidi Exp $

MapServer and OGC Web Services Workshop
=======================================

Contact

Tom Kralidis
Senior Systems Scientist
Environment Canada
+01-416-739-4907
tom.kralidis@ec.gc.ca

INSTALLATION -- UNIX
--------------------

* Note that this UNIX support has not been extensively tested.  If you find any issues, please email me.

1./

- download this zipfile (latest is always at: http://www.maptools.org/ms4w/index.phtml?page=downloads.html)
- unzip package into the root of your file structure, i.e.:

$ lwp-download http://dl.maptools.org/dl/ms-ogc-workshop/ms_ogc_workshop-10.10.zip /tmp/ms_ogc_workshop.zip
$ unzip /tmp/ms_ogc_workshop.zip

The above package will unzip into install into the /ms4w directory structure.

2./ Include the file /ms4w/http.d/httpd_ms_ogc_workshop.conf in your Apache configuration:

i.e. in your httpd.conf file:

include /ms4w/httpd.d/httpd_ms_ogc_workshop.conf

You should then be able to open up the project homepage, at:

http://127.0.0.1/ms_ogc_workshop/index.html


INSTALLATION -- WINDOWS MS4W
----------------------------

1./

You will require the MapServer for Windows (MS4W) package.  The downloads page is at:

http://www.maptools.org/ms4w/index.phtml?page=downloads.html

a./
- download/install the latest ms4w package.

The above package will install MapServer software, as well as an instance of the Apache Web Server.  At this point, you can run C:\ms4w\Apache\bin\Apache.exe, which will start the web server, and point your web browser to http://127.0.0.1/, and you will see the ms4w homepage with some links.

b./
- download the MapServer OGC Web Services Workshop this zip file form http://www.maptools.org/ms4w/index.phtml?page=downloads.html
- Extract the packages below to the root of the drive where your MS4W is installed on (e.g. C:\ )
- make sure "Use Folder Names" is checked


c./
- Start the Apache web server by double clicking \ms4w\Apache\bin\Apache.exe
- Open your web browser (such as Mozilla/Firefox, Opera or Microsort Internet Explorer)

You should then be able to open up the project homepage, at:

http://127.0.0.1/ms_ogc_workshop/index.html


2./

OTHER HELPFUL RESOURCES

If you wish, you can download and print the OGC WMS Cookbook from:

http://www.opengeospatial.org/docs/2003/20031113_wmscookbook.pdf

3./

If you wish, you can download and print the CGDI Best Practices Guide from:

http://www.geoconnections.org/publications/Technical_Manual/Technical_Manual_1.1_e.pdf (EN)
http://www.geoconnections.org/publications/Technical_Manual/Technical_Manual_1.1_f.pdf (FR)

or the Developers' Guide to the CGDI:

http://www.geoconnections.org/publications/Technical_Manual/html_e/cgdiindex.html (EN)
http://www.geoconnections.org/publications/Technical_Manual/html_f/cgdiindex.html

or CGDI online training:

http://www.geoconnections.org/publications/training_manual/e/ (EN)
http://www.geoconnections.org/publications/training_manual/f/index.htm (FR)

If you have any questions or comments, feel free to contact me.


INSTALLATION -- WINDOWS OSGEO4W
-------------------------------

1./

Starting from version 1.0.9, the workshop is available through the osgeo4w installer
(http://wiki.osgeo.org/wiki/OSGeo4W)

Steps are:
 
  - download/install the http://download.osgeo.org/osgeo4w/osgeo4w-setup.exe 
  - select the the OGC Workshop when installing (It is part of the Web_Applications 
    category)
  

WORKSHOP CONTENT
================

NOTES:

- This package is designed for an ms4w based install, which uses the Apache Web Server (http://httpd.apache.org/).  MapServer with Microsoft IIS support is NOT be supported in this package.  You can configure ms4w with IIS support on your own if you wish, however.

OUTLINE
-------

- workshop will be combination of instruction, hands-on training and discussion
- workshop is not bound to outline below, i.e. if there are specific issues of importance throughout the workshop, they will be addressed

- Introduction
 - Web Services
 - Interoperability
- Applying OGC services using MapServer
 - WMS Server (Cascading)
 - WMS Client
 - WFS Server (Cascading)
 - WFS Client
 - SLD (Style layer descriptors and how to optimize their use)
 - Web Map Context Documents
 - Filter Encoding
 - WCS
- Future steps
- open discussion / forum

If you have any questions or comments, feel free to contact me.



HISTORY.txt

$Id: HISTORY.txt,v 1.32 2010-02-03 17:45:53 tkralidi Exp $

MapServer OGC Web Services Workshop Revision History
====================================================

This is a human-readable revision history which will attempt to document required changes for users to migrate from one version of MapServer to the next.  Developers are strongly encouraged to document their changes and their impacts on the users here (please add the most recent changes to the top of the list).

For a complete change history, please see the CVS log comments.

Current Version (HEAD):
----------------------------
- update html templates for MapServer 5.4 changes
- update links to MapServer website according to new structure
- add .qix for data/obs.shp data/rivers.shp
- MapServer RFC56 changes
- add foss4g2009 to data/mums.gml
- add CONFIG keyword to service/config.map
- add foss4g2010 to data/mums.gml
- add AuthorityURL and Identifier support to service/config.map


Version 1.0.10 (2008-09-10):
----------------------------

- Add BBOX element to the GetCoverage  call in the main index.html
- Add ./sos/util/proxy.php to handle AJAX-y requests out of local domain
- update ./sos/DescribeSensor.xml (use sos:procedure instead of sos:SensorId)
- update ./sos/index.html to use proxy
- add more WMS global layer examples

Version 1.0.9 (2008-03-17)
------------------------------
- few modification to prepare the worksop for osgeo4w installation
- added foss4g2007 evaluations to doc/
- added initial SOS examples in ./sos/ and ./data/obs

Version 1.0.8 (2007-11-14)
------------------------------
- changed various demo_init.html files to adjust to rfc31


Version 1.0.7 (2007-08-24)
------------------------------

- added website pointer to index.html
- changed to php_mapscript.dll pointer in all PHP scripts
- added GetLegendGraphic with SLD example and change to pointer to WxS mapscript PHP file
- added data to data/mums.gml
- updated client/webservices/gazetteer.php to geonames.org web service


Version 1.0.6 (2006-10-16)
------------------------------
- added "gml_include_items" "all" to service/config.map for raster GML query output
- fixed all PHP docs to point to php_mapscript_4.10.0


Version 1.0.5 (2006-10-13)
------------------------------

- updated location of workshop URL in doc/ms_ogc_workshop.ppt


Version 1.0.4 (2006-09-08)
------------------------------

- updated for foss4g2006 conference

Version 1.0.3 (2006-08-31)
------------------------------

- changed schema pointers from schemas.opengeospatial.net to schemas.opengis.net
- changed GML version from 3.0.0 to 3.1.1 in data/mums.xsd
- prepared for move to maptools.org

Version 1.0.2 (2006-08-06)
------------------------------

- Added documentation for SOS
- Added initial mapfile hooks for SOS
- added and modified MapServer HOWTO links
- added links for Mapscript OWS services
- fixed projection codes 4236 to 4326


Version 1.0.1 (2006-01-06)
------------------------------

- Fixed corrupt data/rivers files (not imported into CVS correctly)


Version 1.0.0 (2005-12-29)
------------------------------

- Initial packaging as 1.0.0


client/webservices/gazetteer.php
<?php
//
// $Id: gazetteer.php,v 1.2 2007-08-17 16:35:07 tkralidi Exp $
//

$sRequestURL = "http://ws.geonames.org/search?name=" . $_GET["placename"] ;

// the first time this window is opened (i.e. action=init), it should not query the gazetteer
// if the "action" keyword is not in the GET call, and placename is then go ahead
if (! $_GET["action"] && $_GET["placename"]) {

	// instatiate xml object from remote URL
        $geonames = new SimpleXMLElement(file_get_contents($sRequestURL)); 

	// count the number of results
	$gazNumRecs = $geonames->totalResultsCount;
}

?>

<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<html>
	<head>
		<meta http-equiv="Content-Type" content="text/html; charset=utf-8"/>
		<meta http-equiv="pragma" content="no-cache"/>
		<meta http-equiv="Expires" content="Thu, 01 Dec 1994 120000 GMT"/>
		<title>Gazetteer Lookup Results</title>
		<script type="text/javascript" language="JavaScript">
			function setXY() {

				// set the selected placename's geometry

				var tmparr = new Array();
				tmparr     = document.forms[0].gazEntries.options[document.forms[0].gazEntries.selectedIndex].value.split(",");
				document.forms[0].xCoord.value = tmparr[0];
				document.forms[0].yCoord.value = tmparr[1];
			}

			function validatePlacename() {

				// make sure a request doesn't get sent if placename is blank

				if (document.forms[0].placename.value == "") {
					alert("enter a placename");
					return false;
				}
				else {
					return true;
				}
			}

			function zoomToPlacename() {

				// zoom into map on centroid of placename

				if (document.forms[0].xCoord.value == "" || document.forms[0].yCoord.value == "") {
					alert("select a placename");
					return;
				}
				parent.window.opener.document.forms[0].mapxy.value  = document.forms[0].xCoord.value + " " + document.forms[0].yCoord.value;
				parent.window.opener.document.forms[0].buffer.value = 2;
				parent.window.opener.document.forms[0].submit();
				self.close();
			}
		</script>
	</head>
	<body>
		<h2>Gazetteer Lookup Results</h2>
		<hr noshade/>
		<?
			// do some error checking
			// only if this is NOT the first time the page is being called
			// with action=init
			// if the window is opened for the first time
			// don't do error checking yet
			if (! $_GET["action"]) {
				// do some error checking
				if ($geonames->status) {
					$sExceptionMessage = $geonames->status['message'];
		?>

		<center>No results: <?php echo $sExceptionMessage ?></center>

		<?
				}
				else {
		?>

		<center>Successful!</center>

		<?
				}
			}
		?>

		<form name="gazetteerResultsForm" action="gazetteer.php" method="get" onsubmit="javascript:return validatePlacename();">
			<table align="center" border="1">
				<tr>
					<th>placename</th>
					<td><input type="text" name="placename" value="<?php echo $_GET['placename'] ?>"/><input type="submit" name="find" value="Find"/></td>
				</tr>
				<?
					if ((!$_GET["action"]) && ($_GET["placename"])) {
				?>
				<tr>
					<th>results (<?php echo $gazNumRecs ?>)</th>
					<td>
						<select name="gazEntries" onchange="javascript:setXY();" size="5">
<?

foreach ($geonames->geoname as $geoname) {
	$placename = $geoname->name . ", (" . $geoname->countryName . ")";
	$x = $geoname->lng;
	$y = $geoname->lat;
	echo "							<option value=\"$x,$y\">$placename</option>\n";
}

?>
						</select>
					</td>
				</tr>
				<tr>
					<th>x</th>
					<td><input type="text" name="xCoord" value="" readonly/></td>
				</tr>
				<tr>
					<th>y</th>
					<td><input type="text" name="yCoord" value="" readonly/></td>
				</tr>
				<tr>
					<td colspan="2" align="center"><input type="button" name="zoomToPoint" value="Zoom To Point" onclick="javascript:zoomToPlacename();"/></td>
				</tr>

				<tr>
					<td colspan="2" align="center"><a href="<? echo $sRequestURL ?>" target="xmlWin">XML Response</a></td>
				</tr>
				<?
				}
				?>
				<tr>
					<td colspan="2" align="center"><input type="button" name="close" value="Close" onclick="javascript:self.close();"/></td>
				</tr>
			</table>
		</form>
	</body>
</html>
